### PR TITLE
Implementation of the NominatimLocationRepository and LocationViewModel

### PR DESCRIPTION
**Pull request by daferya**
This pull request introduces several new features and dependencies related to location search functionality, including a new repository, ViewModel, and tests. The most important changes include adding dependencies for `okhttp` and `kotlinx.serialization.json`, implementing the `LocationRepository` and `LocationViewModel`, and adding tests for these new components.

### New Dependencies:
* [`app/build.gradle.kts`](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R164-R169): Added `okhttp` and `kotlinx.serialization.json` dependencies.
* [`gradle/libs.versions.toml`](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR8): Added versions and libraries for `okhttp`, `kotlinx-serialization-json`, and `json`. [[1]](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR8) [[2]](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR33) [[3]](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR45-R54) [[4]](diffhunk://#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87dfR103)

### New Features:
* [`app/src/main/java/com/arygm/quickfix/model/locations/LocationRepository.kt`](diffhunk://#diff-b4b5c582e6881624b2b2c352413b176b7e21b63161af8d7a30d3c7aadf0a361bR1-R5): Introduced `LocationRepository` interface for location search functionality.
* [`app/src/main/java/com/arygm/quickfix/model/locations/LocationViewModel.kt`](diffhunk://#diff-5fca28a7e7565e787889496f92eda40aa3e594e60c578624c2a26365c23338cbR1-R44): Implemented `LocationViewModel` to manage search queries and results.
* [`app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt`](diffhunk://#diff-1ee345d2774bb26562a7280f4d00a2a042ccdcc899d816d79be6e7bf3ac4710aR1-R61): Added `NominatimLocationRepository` class to handle location search requests using `okhttp`.

### Testing:
* [`app/src/test/java/com/arygm/quickfix/location/LocationViewModelTest.kt`](diffhunk://#diff-79d9cb8030494e91c33814194755ea6c72c442cc643e243758f6174d3da50ea6R1-R68): Added unit tests for `LocationViewModel`.
* [`app/src/test/java/com/arygm/quickfix/location/NominatimLocationRepositoryTest.kt`](diffhunk://#diff-54baa46c922fbf895cf82a4eb6ee1655a6ccecb5c9e677b21197b189df3d9f51R1-R164): Added unit tests for `NominatimLocationRepository`.

### Configuration:
* [`app/src/main/AndroidManifest.xml`](diffhunk://#diff-7fa6aef292187a049f7a4d6060d8df3ba212d838789c78940bd363344b1c38cdR31-R33): Added metadata for Google Maps API key.

### CODE REVIEW

File: app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      onFailure: (Exception) -> Unit
+  ) {
+    val url = "https://nominatim.openstreetmap.org/search?q=${query}&format=json"
+
+    val request =
+        Request.Builder()
+            .url(url)
+            .header(
+                "User-Agent", "QuickFix/1.0 (daferyassine52@gmail.com)") // Set a proper User-Agent
+            .header("Referer", "https://QuickFix.com") // Optionally add a Referer
```
Comment (AnotherMedo) [inline app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt:20]: we should check this, we do not own the QuickFix domain and putting a personnal email address may not be the best idea

Comment (AnotherMedo) [inline app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt:20]: DNS names are all lowercase:
```suggestion
            .header("Referer", "https://quickfix.com") // Optionally add a Referer
```

**Review by AnotherMedo — CHANGES_REQUESTED** [review]
Not much to say here, very nice work! The code is very clean and concise which made it a treat to read and review 😄 

I just have a few reservations about the use of `https://quickfix.com` and that of a personal email address for the user agent, could you justify that choice ?

**Review by mohagpt — COMMENTED** [review]
The feature added is well-implemented and a great addition, as it can be reused in several classes. I have nothing more to add to the implementation itself. However, I agree with Mehdi's change requests; they are valid and should be addressed before merging this into the main branch. Once those adjustments are made, this will be ready to go! Great work!

File: app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt (hunk 1 above)
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt]: Thank you for pointing this out! I have updated the DNS name to lowercase as suggested  Please let me know if there’s anything else to adjust.

Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt]: Thank you for bringing this up! You're absolutely right. Using a personal email address and referencing a domain we don’t own may not be ideal. I will consult with the team to determine the best approach. Let me know if you have any specific recommendations!

**Pull request merged**